compiler/src/iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp

```diff
@@ -291,23 +299,30 @@ class VulkanSPIRVTargetBackend : public TargetBackend {
     // If we had multiple target environments we would generate one target attr
     // per environment, with each setting its own environment attribute.
     targetAttrs.push_back(getExecutableTarget(
-        context, getSPIRVTargetEnv(options_.vulkanTargetEnv,
-                                   options_.vulkanTargetTriple, context)));
+        context,
+        getSPIRVTargetEnv(options_.targetEnv, options_.targetTriple, context),
+        options_.indirectBindings));
     return ArrayAttr::get(context, targetAttrs);
   }
 
   IREE::HAL::ExecutableTargetAttr
-  getExecutableTarget(MLIRContext *context,
-                      spirv::TargetEnvAttr targetEnv) const {
+  getExecutableTarget(MLIRContext *context, spirv::TargetEnvAttr targetEnv,
+                      bool indirectBindings) const {
     Builder b(context);
     SmallVector<NamedAttribute> configItems;
 
     configItems.emplace_back(b.getStringAttr(spirv::getTargetEnvAttrName()),
                              targetEnv);
+    if (indirectBindings) {
+      configItems.emplace_back(b.getStringAttr("hal.indirect_bindings"),
+                               UnitAttr::get(context));
+    }
 
     auto configAttr = b.getDictionaryAttr(configItems);
     return IREE::HAL::ExecutableTargetAttr::get(
-        context, b.getStringAttr("vulkan"), b.getStringAttr("vulkan-spirv-fb"),
+        context, b.getStringAttr("vulkan"),
+        indirectBindings ? b.getStringAttr("vulkan-spirv-fb")
+                         : b.getStringAttr("vulkan-spirv-fb-ptr"),
```

Comment:
Isn't this flipped? This changes the value with the default flag setting (`false`).